lib/declare_schema/model/foreign_key_definition.rb

```diff
@@ -15,10 +17,10 @@ def initialize(model, foreign_key, options = {})
         @child_table = model.table_name # unless a table rename, which would happen when a class is renamed??
         @parent_table_name = options[:parent_table]&.to_s
         @foreign_key_name = options[:foreign_key]&.to_s || @foreign_key
-        @index_name = options[:index_name]&.to_s || model.connection.index_name(model.table_name, column: @foreign_key_name)
 
-        # Empty constraint lets mysql generate the name
-        @constraint_name = options[:constraint_name]&.to_s || @index_name&.to_s || ''
+        @constraint_name = options[:constraint_name]&.to_s ||
+                             options[:index_name]&.to_s ||
+                             IndexDefinition.index_name(@foreign_key_name)
```

Comment:
The `index_name` method from Rails just generated an index name like `index_users_on_first_name_and_last_name` and we cut down on that verbosity because it was blowing out the MySQL length limit sometimes. (And there's no need to put the table name up front since indexes name scope is always within a table.) We were `gsub` ing before to simplify that, so I decided just to make a class method out of what we really want.